pkg/storegateway/stats.go

```diff
@@ -110,6 +111,11 @@ func newBlockQueriedMeta(meta *block.Meta) blockQueriedMeta {
 	if meta.Compaction.Level == 0 {
 		m.level = "unknown/old_block"
 	}
+
+	// The compactor doesn't preserve the "FromOutOfOrder" hints. Instead, we should check for the external label, that Mimir sets.
```

Comment:
Can this logic live in `block.Meta`? Maybe add a method to `block.Meta` that checks both if the compactor set out-of-order or if the external labels have it? It seems to be entirely determined from block metadata.